Source/Cells.cs

```
@@ -0,0 +1,164 @@
+                    liveCellCount += cells[currentColumn, previousRow] ? (ushort)1 : (ushort)0;
+                    liveCellCount += cells[nextColumn, previousRow] ? (ushort)1 : (ushort)0;
+                    liveCellCount += cells[nextColumn, currentRow] ? (ushort)1 : (ushort)0;
+                    liveCellCount += cells[nextColumn, nextRow] ? (ushort)1 : (ushort)0;
+                    liveCellCount += cells[currentColumn, nextRow] ? (ushort)1 : (ushort)0;
+                    liveCellCount += cells[previousColumn, nextRow] ? (ushort)1 : (ushort)0;
+
+                    bool isCurrentCellAlive = false;
+                    switch (liveCellCount)
+                    {
+                        case 3:
+                            isCurrentCellAlive = true;
+                            break;
+                        case 2:
+                            if (cells[currentColumn, currentRow])
+                            {
+                                isCurrentCellAlive = true;
+                            }
+                            break;
+                    }
+
+                    nextGenerationCells[currentColumn, currentRow] = isCurrentCellAlive;
+                }
+            }
+
+            cells = nextGenerationCells;
+            Thread.Sleep(sleepTime);
+        }
+
+        /// <summary>
+        /// Function that sets specific cells to be alive or dead.
+        /// </summary>
+        /// <param name="isAlive">Status to set.</param>
+        /// <param name="column">Which column.</param>
+        /// <param name="row">Which row.</param>
+        /// <exception cref="IndexOutOfRangeException">If given cell is not present in field.</exception>
+        public void SetCellsAlive(bool isAlive, int column, int row)
+        {
+            if(column < 0 || row < 0 || column > ColumnCount - 1 || row > RowCount - 1)
+            {
+                throw new IndexOutOfRangeException("Cannot set cells outside of bounds!");
```

Is there a need to throw an exception here?
What happens if the exception is thrown?

---

> Is there a need to throw an exception here? What happens if the exception is thrown?

fixed